File: backend/src/main/java/br/com/asa/services/HouseholdService.java
package br.com.asa.services;

import java.util.List;
import java.util.Optional;

import br.com.asa.dtos.HouseholdDto;
import br.com.asa.models.HouseholdEntity;
import org.springframework.data.domain.Page;

public interface HouseholdService {
	Page<HouseholdEntity> findAll(Integer pageNo, Integer pageSize, String sortBy);
	Optional<HouseholdEntity> findById(Integer id);
	List<HouseholdEntity> getByFamilyReferenceName(String search);
	void save(HouseholdDto householdDto);
	List<HouseholdDto> getChildrenByFamilyId(int familyId);
}

File: frontend/src/services/HouseholdService.js
import http from '@/services/http-common'

const path = '/household'

class HouseholdService {
  getAll(currentPage) {
    return http.get(`${path}?pageNo=${currentPage}`)
  }

  getById(id) {
    return http.get(`${path}/view`, id)
  }

  getByFamilyReferenceName(search) {
    return http.get(`${path}/search?familyReference=${search}`)
  }

  new(data) {
    return http.post(`${path}/new`, data)
  }

  edit(data) {
    return http.put(`${path}/edit`, data)
  }

  getChildrenByFamilyId(familyId) {
    //TODO
    familyId = 1
    return http.get(`${path}/view/children/${familyId}`)
  }
}

export default new HouseholdService()
